src/peft/tuners/lora/config.py

```diff
@@ -259,6 +259,27 @@ class LoraConfig(PeftConfig):
             )
         },
     )
+    use_mora: bool = field(
+        default=False,
+        metadata={
+            "help": (
+                "Enable MoRA: High-Rank Updating for Parameter-Efﬁcient Fine-Tuning. This technique employs a square "
+                "matrix to achieve high-rank updating while maintaining the same number of trainable parameters. "
+                "To achieve it, MoRA introduces the corresponding non-parameter operators to reduce the input dimension "
+                "and increase the output dimension for the square matrix. Furthermore, these operators ensure that the "
+                "weight can be merged back into LLMs, which makes the method can be deployed like LoRA."
+            )
+        },
+    )
+    mora_type: int = field(
+        default=1,
+        metadata={
+            "help": (
+                "type 1 (Sharing) for large lora ranks, Eq. 6 in paper. "
+                "type 6 (RoPE based) for small lora ranks, Eq. 9 in paper."
+            )
+        },
+    )
```

Comment:
I would prefer this somehow be an enum, because the original code references possible values for `mora_type` as 1, 2, 3, 4, 6. However only 1 and 6 are ever described anywhere.

Reply:
I think mora_type might be best defined as 3 different flags. where 1, 2, 3,& 4 are permutations of 2 of the flags, and 6 is the rope implementation. @kongds

Reply:
Thanks for your advice.

For MoRA, we only use the `1` and `6` flags.
The flags like `2`, `3`, and `4` are internally used for ReMoRA to change types of sharing groups during training.

Reply:
> I would prefer this somehow be an enum

Even better would be a string enum with some illustrative names. As a user, I would appreciate not having to look up each time what type 1 or type 6 is. The descriptions in the help could also be elaborated.

Reply:
Agreed. I'm not exactly sure what the two flags for the sharing groups would be called though.